lib/rbi/loc.rb

```diff
@@ -31,6 +31,17 @@ def initialize(file: nil, begin_line: nil, end_line: nil, begin_column: nil, end
       @end_column = end_column
     end
 
+    #: (Loc) -> Loc
+    def join(other)
+      Loc.new(
+        file: file,
+        begin_line: begin_line,
+        begin_column: begin_column,
+        end_line: other.end_line,
+        end_column: other.end_column,
+      )
+    end
```

Comment:
Isn't this assuming that `other` will be further up in the document than `self`? What if I pass an `other` with `end_line` that is smaller than `self.begin_line`?

This assumption works for your current use-case, but, IMO, isn't correct for a general purpose method like `join`, especially when the method doesn't do any bounds check to reject invalid values to assert the assumption.